src/probability_theory/stopping.lean

```diff
@@ -306,4 +309,154 @@ end linear_order
 
 end is_stopping_time
 
+section linear_order
+
+
+/-- Given a map `u : ι → α → E`, its stopped process with respect to the stopping
+time `τ` is the map `(i, x) ↦ u (min i (τ x)) x`.
+Intuitively, the stopped process stop evolving once the stopping time has occured. -/
+def stopped_process [linear_order ι] (u : ι → α → β) (τ : α → ι) : ι → α → β :=
+λ i x, u (linear_order.min i (τ x)) x
+
+/-- Given a map `u : ι → α → E`, its stopped value with respect to the stopping
+time `τ` is the map `x ↦ u (τ x) x`. -/
+def stopped_value (u : ι → α → β) (τ : α → ι) : α → β :=
+λ x, u (τ x) x
+
+variable [linear_order ι]
+
+-- We will need cadlag to generalize the following to continuous processes
+section nat
+
+open filtration
+
+variables [hβ : add_comm_monoid β] {f : filtration ℕ m} {u : ℕ → α → β} {τ : α → ℕ}
+
+include hβ
+
+lemma stopped_process_eq (n : ℕ) :
+  stopped_process u τ n =
+  set.indicator (λ a, n ≤ τ a) (u n) +
```

Comment:
there's some defeq abuse going on here; you should rewrite this with the `{|}` syntax